rclcpp/include/rclcpp/client.hpp

```diff
@@ -171,13 +229,20 @@ class ClientBase
   const rcl_node_t *
   get_rcl_node_handle() const;
 
+  RCLCPP_PUBLIC
+  void
+  set_on_new_response_callback(rcl_event_callback_t callback, const void * user_data);
```

Comment:
why is `set_on_new_response_callback` (and the other new ones) protected?

Reply:
Because the `set_on_new_response_callback(std::function<void(size_t)> callback)` signature is the public one?

Does it need to be public?